src/redux/slices/cartSlice.ts

```diff
@@ -0,0 +1,41 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+
+interface Product {
+  id: string;
+  image: string;
+  name: string;
+  price: number;
+}
+
+interface CartProduct {
+  product: Product;
+  quantity: number;
+}
+
+interface CartState {
+  data: {
+    products: CartProduct[];
+    total: number;
+  };
+}
+
```

Comment:
since no interface is being exported, we can make this code readable by creating the one interface